grafsy-client/main.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"time"
+
+	"github.com/leoleovich/grafsy"
+)
+
+var version = "dev"
+
+func main() {
+	var configFile string
+	var connectionTimeout int
+	printVersion := false
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage: %s [args] [file1 [fileN...]]\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "   Or: metrics-generator | %s [args]\n\n", os.Args[0])
+		fmt.Fprintln(os.Stderr, "Reads metrics from files or STDIN and writes to grafsy LocalBind address.")
+		fmt.Fprintf(os.Stderr, "If STDIN contains something, then files will be ignored\n\n")
+		flag.PrintDefaults()
+	}
+	flag.StringVar(&configFile, "c", "/etc/grafsy/grafsy.toml", "Path to config file.")
+	flag.BoolVar(&printVersion, "v", printVersion, "Print version and exit")
+	flag.IntVar(&connectionTimeout, "w", 50, "Timeout ")
+	flag.Parse()
+	args := flag.Args()
+
+	if printVersion {
+		fmt.Printf("Version: %v\n", version)
+		os.Exit(0)
+	}
+
+	var conf grafsy.Config
+	err := conf.LoadConfig(configFile)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+
+	conn, err := net.DialTimeout("tcp", conf.LocalBind, time.Duration(connectionTimeout)*time.Second)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Fail to establish connection: %v\n", err)
+		os.Exit(1)
+	}
+	defer conn.Close()
+
+	stat, err := os.Stdin.Stat()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error in STDIN: %v\n", err)
+		os.Exit(1)
+	}
+
+	// If STDIN is CharDevice, then it contains something.
+	// Send it to grafsy daemon
+	if (stat.Mode() & os.ModeCharDevice) == 0 {
+		stdin := bufio.NewReader(os.Stdin)
+		connWriter := bufio.NewWriter(conn)
+		stdin.WriteTo(connWriter)
+		return
+	}
+
+	if len(args) == 0 {
+		flag.Usage()
+	}
```

Comment:
This name is a bit confusing.
What do you mean by args? Sounds like filenames are args. 
But "usage" kinda says it's something different:
```
Usage: %s [args] [file1 [fileN...]]\n
```

Reply:
Sure, I've adjusted the variable and usage